Classes/BITChannel.m

```diff
@@ -123,39 +128,61 @@ - (BOOL)isQueueBusy {
   return self.channelBlocked;
 }
 
-- (void)persistDataItemQueue {
+- (void)persistDataItemQueue:(char **)string {
   [self invalidateTimer];
-  if (!BITSafeJsonEventsString || strlen(BITSafeJsonEventsString) == 0) {
+  char* jsonStream = NULL;
+  char *newEmptyString = NULL;
+  do {
+    jsonStream = *string;
```

Comment:
Swap this line with the one below it to minimize the opportunities for a race condition which forces the loop to run extra times.